panic: runtime error: invalid memory address or nil pointer dereference

![image](https://github.com/projectdiscovery/nuclei/assets/5724472/4ffee0bd-364b-4f62-834f-e419159cfbfa)

```
[INF] Current nuclei version: v3.1.2 (outdated)
[INF] Current nuclei-templates version: v9.6.6 (outdated)
[WRN] Scan results upload to cloud is disabled.
[INF] New templates added in latest release: 162
[INF] Templates loaded for current scan: 1
[WRN] Executing 1 unsigned templates. Use with caution.
[INF] Targets loaded for current scan: 10000
[INF] Using Interactsh Server: oast.live
[0:00:05] | Templates: 1 | Hosts: 10000 | RPS: 4 | Matched: 0 | Errors: 1 | Requests: 25/10000 (0%)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x13b0566]

goroutine 489 [running]:
github.com/projectdiscovery/nuclei/v3/pkg/operators.(*Operators).GetMatchersCondition(...)
	github.com/projectdiscovery/nuclei/v3/pkg/operators/operators.go:69
github.com/projectdiscovery/nuclei/v3/pkg/operators.(*Operators).Execute(0x0, 0xc003b3ddd0, 0xc003c3d6b0, 0xc003c3d6c0, 0x0)
	github.com/projectdiscovery/nuclei/v3/pkg/operators/operators.go:212 +0x46
github.com/projectdiscovery/nuclei/v3/pkg/protocols/common/interactsh.(*Client).processInteractionForRequest(0xc0014bdce0, 0xc003b9c630, 0xc003b3de00)
	github.com/projectdiscovery/nuclei/v3/pkg/protocols/common/interactsh/interactsh.go:161 +0x26b
github.com/projectdiscovery/nuclei/v3/pkg/protocols/common/interactsh.(*Client).RequestEvent(0xc0014bdce0, {0xc003c3d6f0?, 0x1, 0x2b?}, 0xc003b3de00)
	github.com/projectdiscovery/nuclei/v3/pkg/protocols/common/interactsh/interactsh.go:325 +0x24c
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http.(*Request).ExecuteWithResults.func1.1(0xc002e49b80)
	github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/request.go:401 +0x58b
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http.(*Request).executeRequest(0xc003047340, 0xc0033829d8, 0xc000e09310, 0xc00356cd20, 0x0, 0xc0019a59e8, 0x0?)
	github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/request.go:815 +0x327b
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http.(*Request).ExecuteWithResults.func1({0xc00324c280, 0x265}, 0x10?, 0x49468e?)
	github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/request.go:387 +0x5f7
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http.(*Request).ExecuteWithResults(0xc003047340, 0xc0033829d8, 0xc00356ccf0, 0xc00356cd20, 0xc00356cea0)
	github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/request.go:451 +0x34c
github.com/projectdiscovery/nuclei/v3/pkg/tmplexec/generic.(*Generic).ExecuteWithResults(0xc003063140, 0xc00357e480)
	github.com/projectdiscovery/nuclei/v3/pkg/tmplexec/generic/exec.go:54 +0x296
github.com/projectdiscovery/nuclei/v3/pkg/tmplexec.(*TemplateExecuter).Execute(0xc003214bc0, 0xc00357e480)
	github.com/projectdiscovery/nuclei/v3/pkg/tmplexec/exec.go:148 +0x210
github.com/projectdiscovery/nuclei/v3/pkg/core.(*Engine).executeTemplateWithTargets.func2.1(0xffffffff?, 0xff?, 0xc003032ed0)
	github.com/projectdiscovery/nuclei/v3/pkg/core/executors.go:131 +0x1d8
created by github.com/projectdiscovery/nuclei/v3/pkg/core.(*Engine).executeTemplateWithTargets.func2 in goroutine 107
	github.com/projectdiscovery/nuclei/v3/pkg/core/executors.go:107 +0x4f1
```

--- a/pkg/protocols/common/interactsh/interactsh.go
+++ b/pkg/protocols/common/interactsh/interactsh.go
@@ -151,6 +151,13 @@ func requestShouldStopAtFirstMatch(request *RequestData) bool {
 
 // processInteractionForRequest processes an interaction for a request
 func (c *Client) processInteractionForRequest(interaction *server.Interaction, data *RequestData) bool {
+	var result *operators.Result
+	var matched bool
+	defer func() {
+		if r := recover(); r != nil {
+			gologger.Error().Msgf("panic occurred while processing interaction with result=%v matched=%v err=%v", result, matched, r)
+		}
+	}()
 	data.Event.Lock()
 	data.Event.InternalEvent["interactsh_protocol"] = interaction.Protocol
 	data.Event.InternalEvent["interactsh_request"] = interaction.RawRequest
@@ -158,12 +165,15 @@ func (c *Client) processInteractionForRequest(interaction *server.Interaction, d
 	data.Event.InternalEvent["interactsh_ip"] = interaction.RemoteAddress
 	data.Event.Unlock()
 
-	var result *operators.Result
-	var matched bool
 	if data.Operators != nil {
 		result, matched = data.Operators.Execute(data.Event.InternalEvent, data.MatchFunc, data.ExtractFunc, c.options.Debug || c.options.DebugRequest || c.options.DebugResponse)
 	} else {
-		gologger.Warning().Msgf("No operators found for interactsh interaction %s", interaction.FullId)
+		// this is most likely a bug so error instead of warning
+		var templateID string
+		if data.Event.InternalEvent != nil {
+			templateID = fmt.Sprint(data.Event.InternalEvent[templateIdAttribute])
+		}
+		gologger.Error().Msgf("missing compiled operators for '%v' template", templateID)
 	}
 
 	// for more context in github actions

--- a/pkg/protocols/common/interactsh/interactsh.go
+++ b/pkg/protocols/common/interactsh/interactsh.go
@@ -158,7 +158,13 @@ func (c *Client) processInteractionForRequest(interaction *server.Interaction, d
 	data.Event.InternalEvent["interactsh_ip"] = interaction.RemoteAddress
 	data.Event.Unlock()
 
-	result, matched := data.Operators.Execute(data.Event.InternalEvent, data.MatchFunc, data.ExtractFunc, c.options.Debug || c.options.DebugRequest || c.options.DebugResponse)
+	var result *operators.Result
+	var matched bool
+	if data.Operators != nil {
+		result, matched = data.Operators.Execute(data.Event.InternalEvent, data.MatchFunc, data.ExtractFunc, c.options.Debug || c.options.DebugRequest || c.options.DebugResponse)
+	} else {
+		gologger.Warning().Msgf("No operators found for interactsh interaction %s", interaction.FullId)
+	}
 
 	// for more context in github actions
 	if strings.EqualFold(os.Getenv("GITHUB_ACTIONS"), "true") && c.options.Debug {

--- a/pkg/tmplexec/flow/flow_executor.go
+++ b/pkg/tmplexec/flow/flow_executor.go
@@ -166,7 +166,6 @@ func (f *FlowExecutor) ExecuteWithResults(ctx *scan.ScanContext) error {
 	defer func() {
 		if e := recover(); e != nil {
 			gologger.Error().Label(f.options.TemplateID).Msgf("panic occurred while executing target %v with flow: %v", ctx.Input.MetaInput.Input, e)
-			panic(e)
 		}
 	}()
 

